src/gui/README.md

```diff
@@ -0,0 +1,285 @@
+# Graphical User Interface
+
+The graphical user interface can be access by launching OpenROAD with ``-gui``.
+
+## Commands
+
+### Add buttons to the toolbar
+
+```
+create_toolbar_button [-name name]
+                      -text button_text
+                      -script tcl_script 
+                      [-echo]
+```
+
+Returns: name of the new button, either ``name`` or ``buttonX``.
+
+Options description:
+- ``button_text``: The text to put on the button.
+- ``tcl_script``: The tcl script to evaluate when the button is pressed.
+- ``name``: (optional) name of the button, used when deleting the button.
+- ``echo``: (optional) indicate that the commands in the ``tcl_script`` should be echoed in the log.
+
+To remove the button: 
+
+```
+gui::remove_toolbar_button name
+```
+
+## TCL functions
+
+### Support
+
+Determine is the GUI is active:
+
+```
+gui::enabled
+```
+
+Announce to the GUI that a design was loaded:
+
+```
+gui::design_created
```

Comment:
Note that this is only needed if the design was created through odb api calls and not through read_def or read_db.